ansible-test unit tests fail with pytest>=8

### Summary

When running ansible-test unit tests with pytest>=8 it fails with the following trace. Pinning pytest<8 in requirements.txt fixes the issue.
```
________________________ ERROR collecting test session _________________________
tests/output/.tmp/delegation/python3.9/lib/python3.9/site-packages/pluggy/_hooks.py:501: in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
tests/output/.tmp/delegation/python3.9/lib/python3.9/site-packages/pluggy/_manager.py:119: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
tests/output/.tmp/delegation/python3.9/lib/python3.9/site-packages/_pytest/python.py:211: in pytest_collect_directory
    if pkginit.is_file():
/usr/lib/python3.9/pathlib.py:1456: in is_file
    return S_ISREG(self.stat().st_mode)
/usr/lib/python3.9/pathlib.py:1232: in stat
    return self._accessor.stat(self)
```

### Issue Type

Bug Report

### Component Name

ansible-test

### Ansible Version

```console
$ ansible --version
ansible [core 2.15.9]
  config file = None
  configured module search path = ['/home/ttuffin/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ttuffin/tmp/venv/lib/python3.9/site-packages/ansible
  ansible collection location = /home/ttuffin/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/ttuffin/tmp/venv/bin/ansible
  python version = 3.9.18 (main, Dec 14 2023, 16:09:27) [GCC 13.2.1 20230801] (/home/ttuffin/tmp/venv/bin/python3.9)
  jinja version = 3.1.3
  libyaml = True
$ ansible-test --version
ansible-test version 2.15.9
```


### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
CONFIG_FILE() = None
EDITOR(env: EDITOR) = vim
PAGER(env: PAGER) = less -R
```


### OS / Environment

Arch Linux 

### Steps to Reproduce

1. Install a collection with tests that use the pytest library, e.g. `ansible.eda`:
```
$ ansible-galaxy collection install ansible.eda
```
2. CD to the collection dir:
```
$ cd ~/.ansible/collections/ansible_collections/ansible/eda
```
3. Run ansible-test:
```
$ ansible-test units --venv
```
4. It should fail with the above trace. Now pin pytest<8:
```
$ echo 'pytest<8' >> tests/unit/requirements.txt
```
5. Run ansible-test again. That trace should no longer be present

### Expected Results

ansible-test works with pytest>=8

### Actual Results

ansible-test fails with pytest>=8:
```console
________________________ ERROR collecting test session _________________________
tests/output/.tmp/delegation/python3.9/lib/python3.9/site-packages/pluggy/_hooks.py:501: in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
tests/output/.tmp/delegation/python3.9/lib/python3.9/site-packages/pluggy/_manager.py:119: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
tests/output/.tmp/delegation/python3.9/lib/python3.9/site-packages/_pytest/python.py:211: in pytest_collect_directory
    if pkginit.is_file():
/usr/lib/python3.9/pathlib.py:1456: in is_file
    return S_ISREG(self.stat().st_mode)
/usr/lib/python3.9/pathlib.py:1232: in stat
    return self._accessor.stat(self)
E   PermissionError: [Errno 13] Permission denied: '/home/aur_builder/__init__.py'
```


### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

--- a/changelogs/fragments/ansible-test-pytest-8.yml
+++ b/changelogs/fragments/ansible-test-pytest-8.yml
@@ -1,2 +1,2 @@
 minor_changes:
-  - ansible-test - Add a work-around for permission denied errors when using pytest >= 8 on multi-user systems with an installed version of ansible-test.
+  - ansible-test - Add a work-around for permission denied errors when using ``pytest >= 8`` on multi-user systems with an installed version of ``ansible-test``.

--- a/changelogs/fragments/ansible-test-pytest-8.yml
+++ b/changelogs/fragments/ansible-test-pytest-8.yml
@@ -1,0 +1,2 @@
+minor_changes:
+  - ansible-test - Add a work-around for permission denied errors when using pytest >= 8 on multi-user systems with an installed version of ansible-test.

--- a/test/lib/ansible_test/_internal/commands/units/__init__.py
+++ b/test/lib/ansible_test/_internal/commands/units/__init__.py
@@ -261,6 +261,7 @@ def command_units(args: UnitsConfig) -> None:
             '--junit-xml', os.path.join(ResultType.JUNIT.path, 'python%s-%s-units.xml' % (python.version, test_context)),
             '--strict-markers',  # added in pytest 4.5.0
             '--rootdir', data_context().content.root,
+            '--confcutdir', data_context().content.root,  # avoid permission errors when running from an installed version and using pytest >= 8
         ]  # fmt:skip
 
         if not data_context().content.collection:

